trigger/task.go

```diff
@@ -703,8 +703,11 @@ func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification
 	return t.generate(sub, fmt.Sprintf("changed in runtime by %.1f%% (over threshold of %s%%)", percentChange, percentString), "")
 }
 
-func isFailedTaskStatus(status string) bool {
-	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed || status == evergreen.TaskTestTimedOut
+// isValidFailedTaskStatus only matches task statuses that should be triggered for failure.
+// For example, it excludes  setup failures.
+func isValidFailedTaskStatus(status string) bool {
+	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed ||
```

Comment:
would it make sense to use [TaskNonGenericFailureStatuses](https://github.com/evergreen-ci/evergreen/blob/7fa53940bcca04122f7c23e7fffcdfa37a1139df/globals.go#L314)?